docker/Dockerfile.ubuntu.xenial

```diff
@@ -69,6 +67,23 @@ RUN apt-get update && apt-get -y install \
   softhsm2 \
   valgrind
 
+
+WORKDIR /cmake
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.14.4/cmake-3.14.4.tar.gz -nv
+RUN tar -xzf cmake-3.14.4.tar.gz
+WORKDIR /cmake/cmake-3.14.4
+RUN ./configure
+RUN make -j4
+RUN make install
+
+WORKDIR /boost
+RUN wget https://sourceforge.net/projects/boost/files/boost/1.64.0/boost_1_64_0.tar.gz -nv
+RUN tar -xzf boost_1_64_0.tar.gz
+WORKDIR /boost/boost_1_64_0
+RUN ./bootstrap.sh --prefix=/usr/
+RUN ./b2
+RUN ./b2 install
```

Comment:
This is pretty unfortunate. What do we require from the newer versions?

Reply:
As we discussed in chat currently we have a requirement boost 1.64 only for building tests, for building aktualizr Release we can use min boost version 1.58.
The question is, should we up minimum boost version for the project? 

The minimum possible boost version for us is 1.64

Reply:
Another solution could be to use an alternative to `boost::process` for our tests, maybe get something similar from a third party library. That takes a bit more dev effort but would simplify the CI part.

Reply:
`boost::process` is pretty nice though, just sayin'.

Reply:
I believe it is possible to avoid using boost::process and replace it with analog system call for() exec() etc, but I think we should try to be up to date with the third-party libraries we use, if it's possible.
I believe that the boost::process is not a single feature in the boost 1.64 we can use in further development...

Reply:
Sorry, I don't think I understand your comment. What are you proposing?

Reply:
I think we should use boost 1.64 as a min version.

Reply:
Yeah, I think I agree. It's a bit of a pain, but undoing Mike's rather nice code using `boost::process` seems worse.

Reply:
Ok then we're settled I guess, sorry for derailing

Reply:
No problem, the discussion is still worthwhile, and I was the one who complained originally!

Reply:
test_utils.h::Process (the one that relies on boost::process) can be potentially replaced by Utils::shell() that relies on stdio::popen() if it's really required.